subsys/zigbee/common/zigbee_helpers.c

```diff
@@ -0,0 +1,852 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <ctype.h>
+#include <stddef.h>
+
+#include <dk_buttons_and_leds.h>
+#include <logging/log.h>
+
+#include "zigbee_helpers.h"
+#include <zb_error_handler.h>
+#include <zb_nrf_platform.h>
+#include <zboss_api.h>
+
+
+#ifdef CONFIG_BOARD_NRF52840_PCA10056
+#define ERASE_PIN NRF_GPIO_PIN_MAP(1, 9) /* Pin 1.09. */
+#endif
+#ifdef CONFIG_BOARD_NRF52840_PCA10059
+#define ERASE_PIN NRF_GPIO_PIN_MAP(1, 6) /* Pin 1.06. */
+#endif
+#ifdef CONFIG_BOARD_NRF52833_PCA10100
+#define ERASE_PIN NRF_GPIO_PIN_MAP(1, 6) /* Pin 1.06. */
+#endif
+
+/* Number of retries until the pin value stabilizes. */
+#define READ_RETRIES  10
+
+/* Timeout after which End Device stops to send beacons
+ * if can not join/rejoin a network.
+ */
+#ifndef ZB_DEV_REJOIN_TIMEOUT_MS
+#define ZB_DEV_REJOIN_TIMEOUT_MS (1000 * 200)
+#endif
+
+/* Maximum interval between join/rejoin attempts. */
+#define REJOIN_INTERVAL_MAX_S    (15 * 60)
+
+#define IEEE_ADDR_BUF_SIZE       17
+
+LOG_MODULE_REGISTER(zigbee_helpers, CONFIG_ZIGBEE_HELPERS_LOG_LEVEL);
+
+/* Rejoin-procedure related variables. */
+static zb_uint8_t           stack_initialised             = ZB_FALSE;
+static zb_uint8_t           is_rejoin_procedure_started   = ZB_FALSE;
+static zb_uint8_t           is_rejoin_stop_requested      = ZB_FALSE;
+static zb_uint8_t           is_rejoin_in_progress         = ZB_FALSE;
+static zb_uint8_t           rejoin_attempt_cnt            = 0;
+#if defined ZB_ED_ROLE
+static volatile zb_uint8_t  wait_for_user_input           = ZB_FALSE;
+static volatile zb_uint8_t  is_rejoin_start_scheduled     = ZB_FALSE;
+#endif
+
+/* Forward declarations. */
+static void rejoin_the_network(zb_uint8_t param);
+static void start_network_rejoin(void);
+static void stop_network_rejoin(zb_uint8_t was_scheduled);
+
+/**@brief Function to set the Erase persistent storage
+ *        depending on the erase pin
+ */
+zb_void_t zigbee_erase_persistent_storage(zb_bool_t erase)
+{
+#ifdef ZB_USE_NVRAM
+	zb_set_nvram_erase_at_start(erase);
+#endif
+}
+
+int to_hex_str(char *out, uint16_t out_size, const uint8_t *in,
+	       uint8_t in_size, bool reverse)
+{
+	int bytes_written = 0;
+	int status;
+	int i = reverse ? in_size - 1 : 0;
+
+	for (; in_size > 0; in_size--) {
+		status = snprintf(out + bytes_written,
+				  out_size - bytes_written, "%02x", in[i]);
+		if (status < 0) {
+			return status;
+		}
+
+		bytes_written += status;
+		i += reverse ? -1 : 1;
+	}
+
+	return bytes_written;
+}
+
+int ieee_addr_to_str(char *str_buf, uint16_t buf_len,
+		     const zb_ieee_addr_t addr)
+{
+	return to_hex_str(str_buf, buf_len, (const uint8_t *)addr,
+			  sizeof(zb_ieee_addr_t), true);
+}
+
+uint8_t parse_hex_digit(const char c)
```

Comment:
`parse_hex_digit` replaced with `char2hex` from `sys/util.h`.
Other string-hex functions in this file can also convert in reverse order. They cannot be fully removed because it's not available in their counterparts in `utils.h` but they probably could be much shorter if they used `util.h` functions.
I think it can be postponed to subsequent pull requests